readme.txt
=== WaterWoo PDF Plugin ===
Contributors: littlepackage
Donate link: https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&hosted_button_id=PB2CFX8H4V49L
Tags: book, copyright, digital, ebook, ecommerce, e-commerce, file, marca de agua, pdf, plugin, property, protection, publishing, security, signature, watermark, watermarking, woocommerce
Requires at least: 3.7
Tested up to: 4.1
Stable tag: 1.0.12
License: GPLv3 or later
License URI: http://www.gnu.org/licenses/gpl-3.0.html

Protect your intellectual property! WaterWoo PDF allows WooCommerce site administrators to apply custom watermarks to PDFs upon sale.

== Description ==
WaterWoo PDF is a plugin that adds a watermark to every page of your sold PDF file(s). The watermark is customizable with font face, font color, font size, placement, and text. Not only that, but since the watermark is added when the download button is clicked (either on the customer's order confirmation page or email), the watermark can include customer-specific data such as the customer's first name, last name, and email. Your watermark is highly customizable and manipulatable, practically magic!

**Features:**

* Watermark only designated PDF downloads (as specified by you), or *all* PDF downloads from your site
* Files do not need to be in a specific directory
* Super customizable placement: watermark can be moved all over the page, allowing for different paper sizes (such as letter, A4, legal, etc)
* Watermark is applied to **all** pages of **every** PDF purchased
* Watermarks upon click of either the customer's order confirmation page link or email order confirmation link
* Dynamic customer data inputs (customer first name, last name, email, order paid date, and phone)
* Choice of font face, color, size and placement (horizontal line of text anywhere on the page).

**Premium version:**

[WaterWoo Premium](http://cap.little-package.com/shop/waterwoo-pdf-premium "WaterWoo PDF Premium Version") offers these helpful extra features:

* Watermark all PDF files with same settings OR set individual watermarks per product or even per product variation!
* Supports all versions of Adobe PDF (through 1.7)
* Rotatable transparent page overlay watermark, apart from footer watermark (two watermark locations on the page)
* Additional text formatting options, such as font color and style (bold, italics)
* Additional font (Deja Vu) adding more international character support
* Additional dynamic customer data input (business name, order paid date)
* Begin watermark on selected page of PDF document (to avoid watermarking a cover page, for example)
* Optionally password protect and/or encrypt PDF files
* Optionally prevent copying, annotating, or modifying of your PDF files
* Test watermark and/or manually watermark a file on the fly, from the admin panel

== Installation ==

= To install plugin =
1. Upload the entire "waterwoo-pdf" folder to the "/wp-content/plugins/" directory.
2. Activate the plugin through the "Plugins" menu in WordPress.
3. Visit WooCommerce->Settings->Watermark tab to set your plugin preferences.
4. **Please test your watermarking** by making mock purchases before going live to make sure it works and looks great!
5.  Note: for this to work you need to have pretty URLs enabled from the WP settings. Otherwise a 404 error will be thrown.

= To remove plugin: =

1. Deactivate plugin through the 'Plugins' menu in WordPress

2. Delete plugin through the 'Plugins' menu in WordPress


== Frequently Asked Questions ==
= Where do I change watermark settings? =
You can find the WaterWoo settings page by clicking on the "settings" link under the WaterWoo PDF plugin title on your Wordpress plugins panel, or by navigating to the WooCommerce->Settings->Watermark tab.

= It doesn't work =
First make sure you have checked the box at the top of your settings page (Woocommerce -> Settings -> Watermark) so that watermarking is enabled! Secondly, make sure you have entered your PDF file names correctly in the second field if you've entered any at all. Lastly, WaterWoo may not be able to watermark PDF files version 1.5 and newer. Consider the Premium version of the plugin, which includes the PDF parser necessary for newer files (1.5+).

= How do I test my watermark? =
I recommend creating a coupon in your Woocommerce shop to allow 100% free purchases. Don't share this coupon code with anyone! Test your watermark by purchasing PDFs from your shop using the coupon. It's a bit more tedious. If you want an easier go of it (on-the-fly testing), purchase the Premium version of this plugin.

= Why does the watermark go off the page, create new pages? =
Your watermark text string is too big or long for the page! Try decreasing font size, adjusting rotation, or using the X and Y fine tuners to move the watermark back onto the page. The built-in adjustments on the settings page ultimately allow for watermarking on all document sizes. You may need to edit your watermark if it is too verbose.

= Where do the watermarked files go? =
They are generated with a unique name and stored in the same folder as your original Wordpress/Woo product media upload (usually wp-content/uploads/year/month/file). The unique name includes the order number and a time stamp. If your end user complains of not being able to access their custom PDF for some reason (most often after their max number of downloads is exceeded), you can find it in that folder, right alongside your original.

= Will WaterWoo PDF watermark images? =
WaterWoo PDF is intended to watermark PDF (.pdf) files. If you are specifically looking to watermark image files (.jpg, .jpeg, .gif, .png, .etc), you may want to look into a plugin such as [Image Watermark](http://wordpress.org/plugins/image-watermark/ "Image Watermark Plugin"). WaterWoo PDF will create an watermark on a PDF page but the watermark will NOT sit over image files embedded in the PDF. 

= I get an FPDF error =
If you get the "FPDF error: This document (../../yourfile.pdf) probably uses a compression technique which is not supported by the free parser shipped with FPDI" it is because the PDF you are trying to watermark uses a compression technique not supported by the bundled PDF generator, FPDI. FPDI parses PDFs through version 1.4, and occasionally has troubles with 1.5, 1.6 and 1.7. 

1. Try this [solution using Acrobat](http://stackoverflow.com/a/7155711 "Stack Overflow"), if possible. Alternatively, you can go to Edit->Preflight->Standards and Save As PDF/A.
2. If that doesn't work, test and perhaps purchase the [add-on from SetaSign](http://www.setasign.com/products/fpdi/demos/fpdi-pdf-parser/ "PDF Parser Add-On") and add it into this plugin. That will take some programming chops. Or consider buying the Premium version of this plugin, as it will solve this problem.

= Does this work for ePub/Mobi files =
No. At this time I am unaware of a Woocommerce watermarking plugin for these file types.


== Screenshots ==

1. Screenshot of the settings page, as a Woocommerce settings tab.

== Changelog ==

= 1.0 =
* Initial release

= 1.0.2 =
* Support for landscape orientation

= 1.0.3 = 
* Fixed 4 PHP warnings

= 1.0.4 = 
* Support for odd-sized PDFs

= 1.0.5 = 
* Clean up code in waterwoo-pdf.php class_wwpdf_system_check.php and class_wwpdf_download_product.php
* UTF font encoding
* Support for redirect downloads (as long as file is in wp-content folder)
* Better watermark centering on page

= 1.0.6 =
* Readme updates
* Implemented woo-includes to determine if Woo is active
* Fixed link to settings from plugin page
* Tidy "inc/class_wwpdf_watermark.php"

= 1.0.7 =
* Missing folder replaced

= 1.0.8 =
* Fix default option variable names

= 1.0.9 =
* WC 2.3 ready
* added phone number shortcode
* tidied folder structure

= 1.0.10 = 
* WC 2.3.4 update
* added order paid date shortcode: [DATE]

== Upgrade Notice ==

= 1.0 =
* Initial release

= 1.0.1 =
* Minor changes

= 1.0.2 =
* Support for landscape orientation

= 1.0.3 =
* Fixed 4 PHP warnings

= 1.0.4 = 
* Support for odd-sized PDFs
* Added warning to free users that there may be a delay preparing the free version to work with the upcoming Woocommerce 2.3 release.

= 1.0.5 = 
* UTF font encoding
* Support for redirect downloads (as long as file is in wp-content folder)
* Better watermark centering on page

= 1.0.9 = 
* WC 2.3 ready
* added phone number shortcode

= 1.0.10 = 
* added order paid date shortcode: [DATE]

= 1.0.11 = 
* fix to woo-includes / Woo Dependencies

= 1.0.12 = 
* further fix to woo-includes / Woo Dependencies

changelog.txt
*** Waterwoo PDF ***

2014.10.23 - version 1.0
* Initial release

2014.10.26 - version 1.0.1
* Minor changes

2014.11.30 - version 1.0.3
* Fixed four PHP warnings

2014.12.16 - version 1.0.4
* Support for odd-sized PDFs
* Added warning to free users that there may be a delay preparing the free version to work with the upcoming Woocommerce 2.3 release.

2014.12.29 - version 1.0.5
* Clean up code in waterwoo-pdf.php class_wwpdf_system_check.php and class_wwpdf_download_product.php
* UTF font encoding
* Support for redirect downloads (as long as file is in wp-content folder)
* Better watermark centering on page


2015.1.26 - version 1.0.6
* Readme updates
* Implemented woo-includes to determine if Woo is active
* Fixed link to settings from plugin page
* Tidy "inc/class_wwpdf_watermark.php"

2015.1.26 - version 1.0.7
* Fix missing folder issue

2015.1.27 - version 1.0.8
* Fix default option variable names

2015.2.5 - version 1.0.9
* WC 2.3 ready
* added phone number shortcode
* tidied folder structure

2015.2.17 - version 1.0.10
* WC 2.3.4 update
* added order date shortcode: [DATE]

2015.2.24 - 1.0.11 
* fix to woo-includes / Woo Dependencies

2015.2.25 - 1.0.12
* further fix to woo-includes / Woo Dependencies

waterwoo-pdf.php
<?php
/*
 * Plugin Name: WaterWoo PDF
 * Plugin URI: https://wordpress.org/plugins/waterwoo-pdf/
 * Description: Custom watermark your PDFs upon WooCommerce sale. 
 * Version: 1.0.12
 * Author: Caroline Paquette 
 * Author URI: http://cap.little-package.com/waterwoo-pdf
 * Donate link: https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&hosted_button_id=PB2CFX8H4V49L
 * License: GPLv3 or later
 * License URI: http://www.gnu.org/licenses/gpl-3.0.html
 * 
 * Text Domain: waterwoo-pdf
 * Domain path: /lang
 *
 * Copyright 2013-2015 Caroline Paquette 
 *		
 *     This file is part of WaterWoo PDF, a plugin for WordPress. If
 * 	   it benefits you, please donate to my coffee & pizza fund and/or
 * 	   leave a nice review at:
 * 	   https://wordpress.org/support/view/plugin-reviews/waterwoo-pdf?filter=5
 *     Thank you.
 *
 *     WaterWoo PDF is free software: You can redistribute
 *     it and/or modify it under the terms of the GNU General Public
 *     License as published by the Free Software Foundation, either
 *     version 3 of the License, or (at your option) any later version.
 *     
 *     WaterWoo PDF is distributed in the hope that it will
 *     be useful, but WITHOUT ANY WARRANTY; without even the
 *     implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
 *     PURPOSE. See the GNU General Public License for more details.
 *
 *     You should have received a copy of the GNU General Public License
 *     along with WordPress. If not, see <http://www.gnu.org/licenses/>.
 *
 */

if ( ! defined( 'ABSPATH' ) ) exit;

if ( ! function_exists( 'is_woocommerce_active' ) || ! function_exists( 'get_woocommerce_version' ) ) require_once( 'woo-includes/woo-functions.php' );

if ( ! is_woocommerce_active() ) return;

if ( ! class_exists( 'WaterWooPDF' ) ) :

	class WaterWooPDF {

		/**
		 * @var string
		 */
		public $version = '1.0.12';

		/**
		 * Init
		 */
		public static function init() {
			$wwpdf = new self();
		}

		/**
		 * Constructor
		 */
		public function __construct() {
			
			define( 'WWPDF_BASE_FILE', plugin_basename(__FILE__) );
			define( 'WWPDF_BASE_URL', trailingslashit( plugins_url( 'waterwoo-pdf' ) ) );
			define( 'WWPDF_PATH', plugin_dir_path( __FILE__ ) );
			define( 'WWPDF_VERSION', $this->version );
			define( 'WWPDF_TAB', 'waterwoo-pdf');

			$this->autoload_classes();
			$this->load_hooks();
		
		}
		
		/**
		 * All classes
		 */
		private function plugin_classes() {

			return array(
				'wwpdfwatermark'			=> WWPDF_PATH . 'classes/class_wwpdf_watermark.php',
				'wwpdf_downloadhandler'		=> WWPDF_PATH . 'classes/class_wwpdf_downloadhandler.php',
				'wwpdf_download_handler'	=> WWPDF_PATH . 'classes/class_wwpdf_download_handler.php'
			);

		}

		/**
		 * Autoload classes to reduce memory consumption
		 */

		public function autoload( $class ) {

			$classes = $this->plugin_classes();

			$class_name = strtolower( $class );

			if ( isset( $classes[$class_name] ) && is_readable( $classes[$class_name] ) ) {

				require_once( $classes[$class_name] );

			}

		}

		/**
		 * Load required classes
		 */
		private function autoload_classes() {

			$autoload_is_disabled = defined( 'WWPDF_AUTOLOAD_CLASSES' ) && WWPDF_AUTOLOAD_CLASSES === false;

			if ( function_exists( "spl_autoload_register" ) && ! ( $autoload_is_disabled ) ) {
			   // >= PHP 5.2 - Use auto loading
				if ( function_exists( "__autoload" ) ) {
					spl_autoload_register( "__autoload" ); 
				}
				spl_autoload_register( array( $this, 'autoload' ) );
			} else {
				// < PHP5.2 - Require all classes
				foreach ( $this->plugin_classes() as $id => $path ) {
					if ( is_readable( $path ) && ! class_exists( $id ) ) {
						require_once( $path );
					}
				}
			}

		}

		/**
		 * Remove woocommerce_download_product action hook and replace
		 */
		public function load_hooks() {

			add_action( 'admin_init', array( $this, 'load_admin_hooks' ) );
			add_action( 'admin_init', array( $this, 'nag_ignore' ) );

			if ( class_exists( 'WC_Download_Handler' ) ) {

				if ( isset( $_GET['download_file'] ) && isset( $_GET['order'] ) && isset( $_GET['email'] ) ) {

					remove_action( 'init', array( 'WC_Download_Handler', 'download_product') );

					if ( version_compare( get_woocommerce_version(), '2.2.99', '<=') ) {
						new WWPDF_DownloadHandler();
					} else {
						WWPDF_Download_Handler::init();
					}
				}

			} else { 

				e_( 'Yikes! Your WooCommerce installation seems to be missing the WC_Download_Handler class.' );

			}

		}

		/**
		* Load the localization 
		*/
		public function load_localization() {

			load_plugin_textdomain( 'waterwoo-pdf', false, dirname( plugin_basename( __FILE__ ) ) . '/lang/' );	

		}
	
		/**
		* Load the admin hooks
		*/
		public function load_admin_hooks() {

			if ( is_admin() ) {

				$this->load_localization();

				add_filter( 'plugin_row_meta', array( $this, 'add_support_links' ), 10, 2 );			
				add_filter( 'plugin_action_links_' . WWPDF_BASE_FILE, array( $this, 'add_settings_link') );
				add_filter( 'woocommerce_settings_tabs_array', array( $this, 'add_settings_tab' ), 101  );
				add_action( 'woocommerce_settings_tabs_' . WWPDF_TAB, array( $this, 'create_settings_page' ) ); 
				add_action( 'woocommerce_update_options_' . WWPDF_TAB, array( $this, 'save_settings_page' ) );
				add_action( 'current_screen', array( $this, 'load_screen_hooks' ) );

				$plugin = plugin_basename( __FILE__ );

				add_filter( "plugin_action_links_{$plugin}", array( $this, 'upgrade_to_premium_link' ) );

			}
		}	

		/**
		* Add various support links to plugin page
		*/
		public function add_support_links( $links, $file ) {

			if ( ! current_user_can( 'install_plugins' ) ) {

				return $links;

			}
		
			if ( $file == WWPDF_BASE_FILE ) {
				$links[] = '<a href="http://wordpress.org/extend/plugins/waterwoo-pdf/faq/" target="_blank" title="' . __( 'FAQ', 'waterwoo-pdf' ) . '">' . __( 'FAQ', 'waterwoo-pdf' ) . '</a>';
				$links[] = '<a href="http://wordpress.org/support/plugin/waterwoo-pdf" target="_blank" title="' . __( 'Support', 'waterwoo-pdf' ) . '">' . __( 'Support', 'waterwoo-pdf' ) . '</a>';
			}

			return $links;

		}
		
		/**
		* Add settings link to plugin page
		*/
		public function add_settings_link( $links ) {

			$settings = sprintf( '<a href="%s" title="%s">%s</a>' , admin_url( 'admin.php?page=wc-settings&tab=' . WWPDF_TAB ) , __( 'Go to the settings page', 'waterwoo-pdf' ) , __( 'Settings', 'waterwoo-pdf' ) );
			array_unshift( $links, $settings );
			return $links;

		}

		/**
		* Check if we are on settings page
		*/
		public function is_settings_page() {

			if( isset( $_GET['page'] ) && isset( $_GET['tab'] ) && $_GET['tab'] == WWPDF_TAB ) {

				return true;

			} else {

				return false;

			}

		}

		/**
		 * Load screen hooks
		 */
		public function load_screen_hooks() {

			$screen = get_current_screen();

			if( $this->is_settings_page() ) {

				add_action( 'admin_print_scripts', array( $this, 'add_scripts' ) );
				add_action( 'load-' . $screen->id, array( $this, 'add_help_tabs' ) );
				add_action( 'admin_notices', array( $this, 'wwpdf_admin_notices' ) );

			}
		
		}

		/**
		 * Add the scripts	
		 */
		public function add_scripts() {

			wp_enqueue_script( 'wwpdf-scripts', WWPDF_BASE_URL . '/assets/js/wwpdf_admin.js', 'jquery', WWPDF_VERSION );

		}

		/**
		 * Add the help tabs	
		 */
		public function add_help_tabs() {

			// Check current admin screen
			$screen = get_current_screen();

			// Remove all existing tabs
			$screen->remove_help_tabs();
			
			// Create arrays with help tab titles
			$screen->add_help_tab(array(
				'id' => 'waterwoo-pdf-usage',
				'title' => __( 'About the Plugin', 'waterwoo-pdf' ),
				'content' => 
					'<h3>' . __( 'About WaterWoo PDF', 'waterwoo-pdf' ) . '</h3>' .
					'<p>' . __( 'Protect your intellectual property! WaterWoo PDF allows WooCommerce site administrators to apply custom watermarks to PDFs upon sale.' ). '</p>' . 
					'<p>' . __( 'WaterWoo PDF is a plugin that adds a watermark to every page of your PDF file(s). The watermark is customizable with font face, font color, font size, placement, and text. Not only that, but since the watermark is added when the download button is clicked (either on the customer\'s order confirmation page or email), the watermark can include customer-specifc data such as the customer\'s first name, last name, and email. Your watermark is highly customizable and manipulatable.', 'waterwoo-pdf' ) . '</p>' .
	'<p>' . __( '<a href="http://cap.little-package.com/shop/waterwoo-pdf-premium" target="_blank">Consider upgrading to the Premium version</a> if you need more functionality. The premium version adds the ability to start watermarking on a specified page, password protect and copy/print/modify protect your document, a watermark overlay, HTML input, and UTF-8 and extended Unicode support for many more foreign language characters. Thanks again, and enjoy!', 'waterwoo-pdf' ) . '</p>'

				) );

			// Create help sidebar
			$screen->set_help_sidebar(
				'<p><strong>' . __( 'For more information:', 'waterwoo-pdf' ) . '</strong></p>'.
				'<p><a href="http://wordpress.org/extend/plugins/waterwoo-pdf/faq/" target="_blank">' . __( 'Frequently Asked Questions', 'waterwoo-pdf' ) . '</a></p>' .
				'<p><a href="http://wordpress.org/extend/plugins/waterwoo-pdf/" target="_blank">' . __( 'Project on WordPress.org', 'waterwoo-pdf' ) . '</a></p>' .
				'<p><a href="http://cap.little-package.com/shop/waterwoo-pdf-premium" target="_blank">' . __( 'Upgrade to Premium', 'waterwoo-pdf' ) . '</a></p>'

			);

		}

		/**
		 * Add CTA link on plugins page
		 */

		public function upgrade_to_premium_link( $links ) {

			if ( function_exists( 'is_plugin_active' ) && ! is_plugin_active( 'waterwoo-pdf-premium/waterwoo-pdf-premium.php' ) ) {

				$links[] = '<a href="http://cap.little-package.com/shop/waterwoo-pdf-premium" target="_blank">' . __( 'Upgrade to Premium', 'waterwoo-pdf' ) . '</a>';

			}
		
			return $links;

		}

		/**
		 * Upgrade CTA
		 */

		public function get_premium_cta() {

			if ( function_exists( 'is_plugin_active' ) && ! is_plugin_active( 'waterwoo-pdf-premium/waterwoo-pdf-premium.php' ) ) {

				$link = 'http://cap.little-package.com/shop/waterwoo-pdf-premium';

				$getPremium = "<div style='display: none;' id='wwpdf-screen-options-link-wrap'><div id='contextual-help-link-wrap'><a target='_blank' class='show-settings' href='{$link}'>" .

					__( 'Upgrade to WaterWoo PDF Premium', 'waterwoo-pdf' ) .

					"</a></div></div>";

				echo $getPremium;

			}

		}

		/**
		 * Add a tab to the settings page	
		 */
		public function add_settings_tab($tabs) {

			$tabs[WWPDF_TAB] = __( 'Watermark', 'waterwoo-pdf' );
			return $tabs;

		}

		/**
		 * Display a notice that can be dismissed
		 */ 
		public function wwpdf_admin_notices() {
			global $current_user, $pagenow;
			$user_id = $current_user->ID;
			/* Check that the user hasn't already clicked to ignore the message */
			if ( ! get_user_meta($user_id, 'wwpdf_ignore_notice3') ) {
				$currentscreen = get_current_screen();
				if ( $pagenow == 'admin.php' && $currentscreen->id == 'woocommerce_page_wc-settings' ) {
					echo '<div class="updated"><p>'; 
					printf(__('<strong>Yay!</strong> WaterWoo is now WooCommerce 2.3-ready and has a couple new features!<br />WaterWoo has downloaded well over 600 times, but it has only <a href="https://wordpress.org/support/view/plugin-reviews/waterwoo-pdf?filter=5">five (5-star) ratings</a>, and <em>not a single donation</em>.<br />My friends, <a href="https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&hosted_button_id=PB2CFX8H4V49L">please make a small donation</a> or a submit review to help keep me going.<br /><strong>Thank you</strong> for using WaterWoo!<br /><br /><a href="%1$s">Hide This Notice</a>'), '?page=wc-settings&tab=waterwoo-pdf&nag_ignore=0');
					echo "</p></div>";
				}

			}

		}

		/**
		 * Display a notice that can be dismissed
		 */ 
		public function nag_ignore() {
			global $current_user;
			$user_id = $current_user->ID;
			/* If user clicks to ignore the notice, add that to their user meta */
			if ( isset($_GET['nag_ignore']) && '0' == $_GET['nag_ignore'] ) {
				delete_user_meta($user_id, 'wwpdf_ignore_notice', 'true', true);
				delete_user_meta($user_id, 'wwpdf_ignore_notice2', 'true', true);
				add_user_meta($user_id, 'wwpdf_ignore_notice3', 'true', true);
			}
		}

		/**
		 * Clean old pre-1.0.5 settings from options table
		 */
		private function clean_options_table() {

			delete_site_option( 'enable_wwpdf' );
			delete_site_option( 'pdf_files' );
			delete_site_option( 'footer_input' );
			delete_site_option( 'footer_size' );
			delete_site_option( 'footer_color' );
			delete_site_option( 'footer_finetune_Y' );
			delete_site_option( 'wwpdf_settings' );

		}

		/**
		 * Settings array
		 */
		private function settings_array() {

			$wwpdf_enable_default = get_site_option( 'wwpdf_enable', 'no' );
			$wwpdf_files_default = get_site_option( 'wwpdf_files', sprintf( '', PHP_EOL ) );
			$wwpdf_footer_input_default = get_site_option( 'wwpdf_footer_input', 'Licensed to [FIRSTNAME] [LASTNAME], [EMAIL]' );
			$wwpdf_footer_size_default = get_site_option( 'wwpdf_footer_size', '12' );
			$wwpdf_footer_color_default = get_site_option( 'wwpdf_footer_color', '#000000' );
			$wwpdf_footer_y_default = get_site_option( 'wwpdf_footer_y', '270' );
	
			return array(

				array(
					'id' 		=> 'wwpdf_general_options',
					'type' 		=> 'title',
					'title' 	=> __( 'WaterWoo General Options', 'waterwoo-pdf' ),
					'desc' 		=> '',
				),
	
				array(	
					'id' 		=> 'wwpdf_enable',
					'type' 		=> 'checkbox', 
					'title' 	=> __( 'Enable Watermarking', 'water_woo_pdf' ), 
					'desc' 	=> __( 'Check to enable PDF watermarking', 'waterwoo-pdf' ), 
					'default' 	=> $wwpdf_enable_default,
				),

				array(
					'id' 		=> 'wwpdf_files',
					'type' 		=> 'textarea', 
					'title' 	=> __( 'File(s) to watermark', 'waterwoo-pdf' ), 
					'desc' 		=> __( 'List file name(s) of PDF(s) to watermark, one per line, e.g., <code>upload.pdf</code> or <code>my_pdf.pdf</code> .<br />If left blank, WaterWoo PDF will watermark all PDFs sold through WooCommerce.', 'waterwoo-pdf' ), 
					'default' 	=>  $wwpdf_files_default,
					'css' 		=> 'min-width:600px;',
				),

				array(
					'id' 		=> 'wwpdf_footer_input',
					'type'		=> 'textarea',
					'title' 	=> __( 'Custom text for footer watermark', 'waterwoo-pdf' ),
					'desc' 		=> __( 'Shortcodes available, all caps, in brackets: <code>[FIRSTNAME]</code> <code>[LASTNAME]</code> <code>[EMAIL]</code> <code>[PHONE]</code> <code>[DATE]</code>', 'waterwoo-pdf' ),
					'default' 	=> $wwpdf_footer_input_default,
					'css' 		=> 'min-width:600px;',
				),
	
				array(	
					'id' 		=> 'wwpdf_font',
					'type' 		=> 'select',
					'title' 	=> __( 'Font face', 'waterwoo_pdf' ),
					'desc' 	=> __( 'Select a font for watermarks.', 'waterwoo-pdf' ),
					'css' 		=> 'min-width:300px;',
					'class'		=> 'chosen_select',
					'default'	=> 'helvetica',
					'options' => array(
							'helvetica'  => 'Helvetica',
							'times'   => 'Times New Roman',				
							'courier' => 'Courier',
							),
					'desc_tip' 	=> true,

				),

				array(
					'id' 		=> 'wwpdf_footer_size',
					'type' 		=> 'number',
					'title' 	=> __( 'Font size', 'waterwoo-pdf' ),
					'css' 		=> 'width:50px;',
					'desc' 		=> __( 'Provide a number (suggested 10-20) for the footer watermark font size', 'waterwoo-pdf' ),
					'default' 	=> $wwpdf_footer_size_default,
					'custom_attributes' => array(
									'min' 	=> 8,
									'max' => 22,
									'step' 	=> 1,
								),
					'desc_tip' 	=> true,
				),

				array(
					'id' 		=> 'wwpdf_footer_color',
					'type' 		=> 'color',
					'title' 	=> __( 'Watermark color', 'waterwoo-pdf' ),
					'desc' 		=> __( 'Color of the footer watermark. Default is black: <code>#000000</code>.', 'waterwoo-pdf' ),
					'css' 		=> 'width:6em;',
					'default'	=> $wwpdf_footer_color_default,
					'desc_tip' 	=> true,
				),

	
				array(		
					'id' 		=> 'wwpdf_footer_y',
					'type' 		=> 'number',
					'css' 		=> 'width:50px;',
					'title' 	=> __( 'Y Fine Tuning', 'waterwoo-pdf' ),
					'desc' 	=> __( 'Move the footer watermark up and down on the page by adjusting this number. Default is 270 (bottom of page).', 'waterwoo-pdf' ),
					'default' 	=> $wwpdf_footer_y_default,
					'custom_attributes' => array(
									'min' 	=> 0,
									'max' => 1000,
									'step' 	=> 1,
								),
					'desc_tip' 	=> true,
				),

				array( 'id' => 'wwpdf_general_options', 'type' => 'sectionend' ),

			);

		}

		/**
		 * Include and display the settings page.
		 */
		public function create_settings_page() {

			$this->get_premium_cta();

			$waterwoopdf_settings = $this->settings_array();
			woocommerce_admin_fields( $waterwoopdf_settings );		

		}

		/**
		 * Save the settings page.
		 */
		public function save_settings_page() {

			$waterwoopdf_settings = $this->settings_array();		

			if ( is_admin() ) {

				if ( isset( $waterwoopdf_settings ) ) {
					woocommerce_update_options( $waterwoopdf_settings );
				}

				if ( version_compare( WWPDF_VERSION, '1.0.5', '>=') ) {
					// Test if a old value is in database
					$wwpdf_check = get_site_option( 'enable_wwpdf', '', false );

					if ( $wwpdf_check != false ) {
						$this->clean_options_table();
					}
				}

			}

		}

	}

endif;

add_action( 'plugins_loaded', array( 'WaterWooPDF', 'init' ), 10 );

?>